### ISSUE TITLE
Add type tags to `napi-sys`.

### ISSUE THREAD

**Issue by cutsoy** [napi-rs/napi-rs#1491]
Would it be possible to add the following two functions to `napi-sys`? Added since N-API v8 (in Node v14.8.0, v12.19.0).

```c
napi_status napi_type_tag_object(napi_env env,
                                 napi_value js_object,
                                 const napi_type_tag* type_tag);
 
napi_status napi_check_object_type_tag(napi_env env,
                                       napi_value js_object,
                                       const napi_type_tag* type_tag,
                                       bool* result);
```

I'm using the `napi::sys` for some FFI-safe bindings (long story), so if these land in `napi::sys` that would be enough for me. I can submit a PR for this.

---

I can also imagine it would be useful to use this in `TaggedObject` on v8 since it's a bit safer. E.g. the following example segfaults right now:

1. Assume a JS script like this:
   ```js
   const { one } = require("objc-module");
   const { sum } = require("napi-rs-module");
   
   sum(one(), one());
   ```
2. The Objective-C library calls `napi_create_external(...)` with `data = [NSNumber numberWithInt:1]`. This is a [tagged pointer][1], so it doesn't actually point to anything but it also doesn't violate the NAPI contract.
3. The Rust library segfaults because `napi-rs::JsExternal` [attempts to read][2] the first `sizeof(TypeId)` bytes at the tagged pointer's address.

Granted, this example isn't supposed to "work" anyway, but I think it would be better to have it throw an exception instead of segfault.

---

Anyway, summarizing: would it be okay if I submit a PR adding these two functions to `napi::sys`?

[1]: https://stackoverflow.com/questions/20362406/tagged-pointers-in-objective-c
[2]: https://github.com/napi-rs/napi-rs/blob/25cc07b476d2f924aa536a13223994ee59892978/crates/napi/src/bindgen_runtime/js_values/external.rs#L68

**Comment by forehalo** [comment]
> Would it be possible to add the following two functions to napi-sys? Added since N-API v8 (in Node v14.8.0, v12.19.0).

Yeah, we should definitely keep `napi-sys` synchronized with the official node-api

> Anyway, summarizing: would it be okay if I submit a PR adding these two functions to napi::sys

Of course you could, and PR is welcome. Only one thing need to be kept in mind, which is putting the new apis under corresponding napi verison feature
